app/controllers/v0/form21p530a_controller.rb

```diff
@@ -73,5 +62,63 @@ def handle_pdf_generation_error(error)
     def stats_key
       'api.form21p530a'
     end
+
+    def transform_country_codes(payload)
+      parsed = JSON.parse(payload)
+      address = parsed.dig('burialInformation', 'recipientOrganization', 'address')
+      if address&.key?('country')
+        transformed_country = extract_country(address)
+        if transformed_country
+          validate_country_code!(transformed_country)
+          address['country'] = transformed_country
+        end
+      end
+      parsed.to_json
+    end
+
+    def validate_country_code!(country_code)
+      return if country_code.blank?
+
+      IsoCountryCodes.find(country_code)
+    rescue IsoCountryCodes::UnknownCodeError
+      claim = SavedClaim::Form21p530a.new
+      claim.errors.add '/burialInformation/recipientOrganization/address/country',
+                       "'#{country_code}' is not a valid country code"
+      raise Common::Exceptions::ValidationErrors, claim
+    end
```

Comment:
These will need to be moved into a model validator and out of the controller. Ticket has been created for this work - part of post-MVP improvements.

https://github.com/department-of-veterans-affairs/va.gov-team/issues/124779